diff --git a/src/deserializer.rs b/src/deserializer.rs
index c7c3217..b4c0032 100644
--- a/src/deserializer.rs
+++ b/src/deserializer.rs
@@ -426,8 +426,8 @@ impl<'a, 'de: 'a, T: DeRecord<'de>> Deserializer<'de>
         self,
         visitor: V,
     ) -> Result<V::Value, Self::Error> {
-        self.next_field_bytes()
-            .and_then(|f| visitor.visit_byte_buf(f.to_vec()))
+        self.next_field()
+            .and_then(|f| visitor.visit_byte_buf(f.as_bytes().to_vec()))
     }
 
     fn deserialize_option<V: Visitor<'de>>(
